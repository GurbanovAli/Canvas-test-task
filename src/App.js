import React from 'react';
import Download from './components/Download';
import Control from './components/ControlFile';
import Input from './components/InputAndOutputFiles/Input';
import Output from './components/InputAndOutputFiles/Output';
import Image from './components/Image';



const App =()=>(
     <Control>
       <main id='main'>
          <h1>Canvas for The Codex</ h1>
          <div className='main_block'>
             <div className='button_block'>
               <Input />
               <Download />
               <Image />
             </div>
             <Output />
          </div>
       </main>
     </Control>
    );

export default App;
