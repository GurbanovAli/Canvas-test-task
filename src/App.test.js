import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

jest.mock('./file.worker.js');

test('<App /> spec', () => {
  const {container} = render(<App />);
  expect(container).toMatchSnapshot();
});
