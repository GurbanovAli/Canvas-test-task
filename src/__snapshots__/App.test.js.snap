// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> spec renders the component 1`] = `
<div>
   
  <main
    id="main"
  >
    <div
      id="container"
    >
      Canvas
      <div
        id="flip"
      >
        <div>
          <div>
            Make
          </div>
        </div>
        <div>
          <div>
            lifeStyle
          </div>
        </div>
        <div>
          <div>
            Awesome!
          </div>
        </div>
      </div>
      for The Codex
    </div>
    <div
      class="main_block"
    >
      <div
        class="button_block"
      >
        <div
          class="button download_button"
        >
          <input
            accept=".txt"
            class="component"
            id="file-input"
            type="file"
          />
          <label
            for="file-input"
          >
              Upload Canvas  
          </label>
        </div>
        <div
          class="download_button"
        >
          <button
            class="button"
          >
            Download canvas
          </button>
        </div>
        <div>
          <button
            class="button download_button"
            id="image"
          >
             Change Background 
          </button>
        </div>
      </div>
      <div
        class="output"
      >
        <pre>
           
          No file
           
        </pre>
      </div>
    </div>
  </main>
   
</div>
`;

exports[`renders learn react link 1`] = `
<div
  id="container"
>
  Canvas
  <div
    id="flip"
  >
    <div>
      <div>
        Make
      </div>
    </div>
    <div>
      <div>
        lifeStyle
      </div>
    </div>
    <div>
      <div>
        Awesome!
      </div>
    </div>
  </div>
  for The Codex
</div>
`;
